src/nile/core/account.py

```diff
@@ -47,32 +46,32 @@ def deploy(self):
 
         return address, index
 
-    def send(self, to, method, calldata):
+    def send(self, to, method, calldata, nonce=None):
         """Execute a tx going through an Account contract."""
         target_address, _ = next(deployments.load(to, self.network)) or to
+        calldata = [int(x) for x in calldata]
 
-        message, signature = self.signer.sign_message(
-            sender=self.address,
-            to=target_address,
-            selector=method,
-            calldata=calldata,
-            nonce=self.get_nonce(),
+        if nonce is None:
+            nonce = int(
+                call_or_invoke(self.address, "call", "get_nonce", [], self.network)
+            )
+
+        (call_array, calldata, sig_r, sig_s) = self.signer.sign_transaction(
+            sender=self.address, calls=[[target_address, method, calldata]], nonce=nonce
         )
 
-        call_or_invoke(
+        params = []
+        params.append(str(len(call_array)))
+        params.extend([str(elem) for sublist in call_array for elem in sublist])
+        params.append(str(len(calldata)))
+        params.extend([str(param) for param in calldata])
+        params.append(str(nonce))
```

Comment:
I don't like this part at all :(

Reply:
Hmm I think this would look cleaner with a helper method to abstract away `str(len())` and `str()` + iteration. It's only 6 lines though, so maybe not worth it